File: LeetCode/283_Move_Zeroes/src/java/Solution.java
package java;

class Solution {
    public void moveZeroes(int[] nums) {

        if (nums == null || nums.length <= 0) {
            return;
        }

        for (int i = 0, j = 0; i < nums.length; i++) {
            if (nums[i] != 0) {
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
                j++;
            }
        }
    }
}

File: LeetCode/283_Move_Zeroes/src/kotlin/Solution.kt
class Solution {
    fun moveZeroes(nums: IntArray): Unit {
        if (nums.isEmpty()) {
            return
        }

        var j: Int = 0
        for (i in 0..nums.size - 1) {
            if (nums[i] != 0) {
                val temp: Int = nums[i]
                nums[i] = nums[j]
                nums[j] = temp
                j++
            }
        }
    }
}